File: backend_ddd/app/Models/BusLocation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class BusLocation extends Model
{
    protected $fillable = ['bus_id', 'latitude', 'longitude'];

    public function bus(): BelongsTo
    {
        return $this->belongsTo(Bus::class);
    }
}

File: front_ddd/lib/features/buses/domain/entities/bus_location.dart
import 'package:equatable/equatable.dart';

class BusLocation extends Equatable {
  final int id;
  final int busId;
  final double latitude;
  final double longitude;
  final DateTime updatedAt;

  const BusLocation({
    required this.id,
    required this.busId,
    required this.latitude,
    required this.longitude,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [id, busId, latitude, longitude, updatedAt];
}
